data/transactions/logic/assembler.go

```diff
@@ -1143,6 +1143,49 @@ func (ops *OpStream) assemble(fin io.Reader) error {
 	return nil
 }
 
+func (ops *OpStream) pragma(line string) error {
+	fields := strings.Split(line, " ")
+	if fields[0] != "#pragma" {
+		return ops.errorf("invalid syntax: %s", fields[0])
+	}
+	if len(fields) < 2 {
+		return ops.error("empty pragma")
+	}
+	key := fields[1]
+	switch key {
+	case "version":
+		if len(fields) < 3 {
+			return ops.error("no version value")
+		}
+		value := fields[2]
+		var ver uint64
+		if ops.pending.Len() > 0 {
+			return ops.error("#pragma version is only allowed before instructions")
+		}
+		ver, err := strconv.ParseUint(value, 0, 64)
+		if err != nil {
+			return ops.error(err)
```

Comment:
should this have a nicer comment? `("bad version number %#v: %v", value, err)` or so?

Reply:
Agreed, I think I'll leave out the strconv error entirely, in fact.  We have a lot of these in existing errors.  If make similar changes as I go through teal3 updates.